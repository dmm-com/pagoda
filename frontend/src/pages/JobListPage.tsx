import ReplayIcon from "@mui/icons-material/Replay";
import { Box, Button, Container, Typography } from "@mui/material";
import { FC, useMemo } from "react";
import { useLocation } from "react-router";
import { useToggle } from "react-use";

import { useAsyncWithThrow } from "../hooks/useAsyncWithThrow";

import { AironeLink } from "components";
import { AironeBreadcrumbs } from "components/common/AironeBreadcrumbs";
import { Loading } from "components/common/Loading";
import { PageHeader } from "components/common/PageHeader";
import { PaginationFooter } from "components/common/PaginationFooter";
import { JobList } from "components/job/JobList";
import { usePage } from "hooks/usePage";
import { aironeApiClient } from "repository/AironeApiClient";
import { topPath } from "routes/Routes";
import { JobListParam } from "services/Constants";

export const JobListPage: FC = () => {
  const location = useLocation();

  const { page, changePage } = usePage();
  const [refresh, toggleRefresh] = useToggle(false);

  const { targetId } = useMemo(() => {
    const params = new URLSearchParams(location.search);
    const targetId = params.get("target_id");
    return {
      targetId: targetId ? Number(targetId) : undefined,
    };
  }, [location.search]);

  const jobs = useAsyncWithThrow(async () => {
    return await aironeApiClient.getJobs(page, targetId);
  }, [page, targetId, refresh]);

  return (
    <Box className="container-fluid">
      <AironeBreadcrumbs>
        <Typography component={AironeLink} to={topPath()}>
          Top
        </Typography>
        <Typography color="textPrimary">ジョブ一覧</Typography>
      </AironeBreadcrumbs>

      <PageHeader title="ジョブ一覧">
        <Button
          variant="outlined"
          color="success"
          onClick={() => toggleRefresh()}
        >
          <ReplayIcon /> ジョブ一覧を更新
        </Button>
      </PageHeader>

      {jobs.loading ? (
        <Loading />
      ) : (
        <Container>
          <JobList jobs={jobs.value?.results ?? []} />
          <PaginationFooter
            count={jobs.value?.count ?? 0}
            maxRowCount={JobListParam.MAX_ROW_COUNT}
            page={page}
            changePage={changePage}
          />
        </Container>
      )}
    </Box>
  );
};
