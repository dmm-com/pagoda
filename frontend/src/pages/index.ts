// FIXME rethink export scope

export * from "./EntryCopyPage";
export * from "./EntryEditPage";
export * from "./EntryEditPage";
export * from "./EntryListPage";

export * from "./UnavailableErrorPage";
