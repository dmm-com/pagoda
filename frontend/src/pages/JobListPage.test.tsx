/**
 * @jest-environment jsdom
 */

import { render, screen, waitFor } from "@testing-library/react";
import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";

import { TestWrapper } from "TestWrapper";
import { JobListPage } from "pages/JobListPage";

const server = setupServer(
  // getRoles
  http.get("http://localhost/job/api/v2/jobs", () => {
    return HttpResponse.json({
      count: 2,
      next: null,
      previous: null,
      results: [
        {
          id: 1,
          operation: 1,
          status: 1,
          passed_time: 1,
          created_at: new Date("2022-01-01T09:00:00.000000+09:00"),
          text: "note",
          target: {
            id: 1,
            name: "target1",
            schema_id: null,
            schema_name: null,
          },
        },
        {
          id: 2,
          operation: 2,
          status: 2,
          passed_time: 2,
          created_at: new Date("2022-01-01T09:00:00.000000+09:00"),
          text: "note",
          target: {
            id: 2,
            name: "target2",
            schema_id: null,
            schema_name: null,
          },
        },
      ],
    });
  }),
);

beforeAll(() => server.listen());

afterEach(() => server.resetHandlers());

afterAll(() => server.close());

describe("JobListPage", () => {
  test("should match snapshot", async () => {
    const result = render(<JobListPage />, {
      wrapper: TestWrapper,
    });
    await waitFor(() => {
      expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
    });

    expect(result).toMatchSnapshot();
  });
});
