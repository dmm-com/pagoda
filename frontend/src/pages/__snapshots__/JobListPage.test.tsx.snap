// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobListPage should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1baykhj"
        >
          <div
            class="MuiBox-root css-t0pdf1"
          >
            <div
              class="MuiBox-root css-jqffe8"
            >
              <nav
                aria-label="breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-1rk5fdy-MuiTypography-root"
                      data-discover="true"
                      href="/ui/"
                    >
                      Top
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1bgbmvw-MuiTypography-root"
                    >
                      ジョブ一覧
                    </p>
                  </li>
                </ol>
              </nav>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1cs9h4l"
        >
          <div
            class="MuiBox-root css-1519avi"
          >
            <div
              class="MuiBox-root css-9rtbay"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2toz8d-MuiTypography-root"
                id="title"
              >
                ジョブ一覧
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
                id="description"
              />
              <div
                class="MuiBox-root css-hoapfw"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSuccess css-deqglb-MuiButtonBase-root-MuiButton-root"
                  id=":r0:"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="ReplayIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8"
                    />
                  </svg>
                   ジョブ一覧を更新
                </button>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-lhgpb-MuiDivider-root"
            />
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
        >
          <table
            class="MuiTable-root css-1xwxv7r-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1msocm9-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-543kl5-MuiTableCell-root"
                  scope="col"
                >
                  対象モデル
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h7gopu-MuiTableCell-root"
                  scope="col"
                >
                  状況
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1fqh6c1-MuiTableCell-root"
                  scope="col"
                >
                  操作
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-vxd1oc-MuiTableCell-root"
                  scope="col"
                >
                  実行時間
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-543kl5-MuiTableCell-root"
                  scope="col"
                >
                  実行日時
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1nttwnn-MuiTableCell-root"
                  scope="col"
                >
                  備考
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root css-18kj4v8-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-1rk5fdy-MuiTypography-root"
                    data-discover="true"
                    href="/ui/entities/0/entries"
                  />
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <div
                      class="MuiBox-root css-1p1m4ay"
                    >
                      <div
                        class="MuiBox-root css-t0xp56"
                      />
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                      >
                        処理前
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError css-1fn4qev-MuiButtonBase-root-MuiButton-root"
                        id=":r1:"
                        tabindex="0"
                        type="button"
                      >
                        再実行
                      </button>
                      <div
                        class="MuiBox-root css-0"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-v8e7gc-MuiButtonBase-root-MuiButton-root"
                          id=":r2:"
                          tabindex="0"
                          type="button"
                        >
                          キャンセル
                        </button>
                      </div>
                    </div>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    作成
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    1
                     s
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    2022/1/1 0:0:0
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    note
                  </p>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-18kj4v8-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-1rk5fdy-MuiTypography-root"
                    data-discover="true"
                    href="/ui/entities/0/entries"
                  />
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <div
                      class="MuiBox-root css-1p1m4ay"
                    >
                      <div
                        class="MuiBox-root css-1p434bd"
                      />
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                      >
                        完了
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    />
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    編集
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    2
                     s
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    2022/1/1 0:0:0
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                  >
                    note
                  </p>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="MuiBox-root css-5hgm98"
            id="pagination"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            >
              1 - 2 / 2 件
            </p>
            <div
              class="MuiStack-root css-1sazv7p-MuiStack-root"
            >
              <nav
                aria-label="pagination navigation"
                class="MuiPagination-root MuiPagination-text css-1xra8g6-MuiPagination-root"
              >
                <ul
                  class="MuiPagination-ul css-1c5o7vy-MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateBeforeIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-current="page"
                      aria-label="page 1"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                      tabindex="0"
                      type="button"
                    >
                      1
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateNextIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-1baykhj"
      >
        <div
          class="MuiBox-root css-t0pdf1"
        >
          <div
            class="MuiBox-root css-jqffe8"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-5cvx94-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-1uwp4ue-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-1rk5fdy-MuiTypography-root"
                    data-discover="true"
                    href="/ui/"
                  >
                    Top
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-jhr02i-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1bgbmvw-MuiTypography-root"
                  >
                    ジョブ一覧
                  </p>
                </li>
              </ol>
            </nav>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1cs9h4l"
      >
        <div
          class="MuiBox-root css-1519avi"
        >
          <div
            class="MuiBox-root css-9rtbay"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2toz8d-MuiTypography-root"
              id="title"
            >
              ジョブ一覧
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
              id="description"
            />
            <div
              class="MuiBox-root css-hoapfw"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSuccess MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSuccess css-deqglb-MuiButtonBase-root-MuiButton-root"
                id=":r0:"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ReplayIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8"
                  />
                </svg>
                 ジョブ一覧を更新
              </button>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-lhgpb-MuiDivider-root"
          />
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-5c1adp-MuiContainer-root"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1msocm9-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-543kl5-MuiTableCell-root"
                scope="col"
              >
                対象モデル
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-h7gopu-MuiTableCell-root"
                scope="col"
              >
                状況
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1fqh6c1-MuiTableCell-root"
                scope="col"
              >
                操作
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-vxd1oc-MuiTableCell-root"
                scope="col"
              >
                実行時間
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-543kl5-MuiTableCell-root"
                scope="col"
              >
                実行日時
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1nttwnn-MuiTableCell-root"
                scope="col"
              >
                備考
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-18kj4v8-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body1 css-1rk5fdy-MuiTypography-root"
                  data-discover="true"
                  href="/ui/entities/0/entries"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <div
                  class="MuiBox-root css-j7qwjs"
                >
                  <div
                    class="MuiBox-root css-1p1m4ay"
                  >
                    <div
                      class="MuiBox-root css-t0xp56"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                    >
                      処理前
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-0"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorError css-1fn4qev-MuiButtonBase-root-MuiButton-root"
                      id=":r1:"
                      tabindex="0"
                      type="button"
                    >
                      再実行
                    </button>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-v8e7gc-MuiButtonBase-root-MuiButton-root"
                        id=":r2:"
                        tabindex="0"
                        type="button"
                      >
                        キャンセル
                      </button>
                    </div>
                  </div>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  作成
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  1
                   s
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  2022/1/1 0:0:0
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  note
                </p>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-18kj4v8-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <a
                  class="MuiTypography-root MuiTypography-body1 css-1rk5fdy-MuiTypography-root"
                  data-discover="true"
                  href="/ui/entities/0/entries"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <div
                  class="MuiBox-root css-j7qwjs"
                >
                  <div
                    class="MuiBox-root css-1p1m4ay"
                  >
                    <div
                      class="MuiBox-root css-1p434bd"
                    />
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                    >
                      完了
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-0"
                  />
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  編集
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  2
                   s
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  2022/1/1 0:0:0
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
                >
                  note
                </p>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="MuiBox-root css-5hgm98"
          id="pagination"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            1 - 2 / 2 件
          </p>
          <div
            class="MuiStack-root css-1sazv7p-MuiStack-root"
          >
            <nav
              aria-label="pagination navigation"
              class="MuiPagination-root MuiPagination-text css-1xra8g6-MuiPagination-root"
            >
              <ul
                class="MuiPagination-ul css-1c5o7vy-MuiPagination-ul"
              >
                <li>
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateBeforeIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                  </button>
                </li>
                <li>
                  <button
                    aria-current="page"
                    aria-label="page 1"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                    tabindex="0"
                    type="button"
                  >
                    1
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-colorPrimary MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1bug3cd-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-8uzxat-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateNextIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
